### PR TITLE
CIRC-842 add loans fetching for session records

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Approach
- add loans fetching for session records;
- update tests;

Resolves: [CIRC-842](https://issues.folio.org/browse/CIRC-842)
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

### CODE REVIEW

File: src/main/java/org/folio/circulation/infrastructure/storage/sessions/PatronActionSessionRepository.java [hunk 1]
```diff
@@ -151,6 +156,7 @@ private JsonObject mapToJson(PatronSessionRecord patronSessionRecord) {
     return findWithMultipleCqlIndexValues(patronActionSessionsStorageClient,
       PATRON_ACTION_SESSIONS, PatronSessionRecord::from)
       .findByIdIndexAndQuery(patronIds, PATRON_ID, actionTypeQuery)
+      .thenCompose(r -> r.after(this::fetchLoans))
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/infrastructure/storage/sessions/PatronActionSessionRepository.java:159]: Do the endpoints that execute this code have the appropriate permissions to fetch loans?

Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/infrastructure/storage/sessions/PatronActionSessionRepository.java:159]: Yes, the endpoint `/circulation/notice-session-expiration-by-timeout` already has permission for `circulation-storage.loans.collection.get`

Comment (marcjohnson-kint) [reply src/main/java/org/folio/circulation/infrastructure/storage/sessions/PatronActionSessionRepository.java:159]: Does `fetchLoans` only fetch the loan, or does it also fetch related records like items etc?

Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/infrastructure/storage/sessions/PatronActionSessionRepository.java:159]: It does fetch loan with item and loanPolicy

File: src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionService.java [hunk 2]
```diff
@@ -147,6 +152,7 @@ public PatronActionSessionService(
       .collect(Collectors.toList());
 
     if (sessionRecordsWithLoans.isEmpty()) {
+      log.info("Loans were not fetched for the records. The notice will not be sent");
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionService.java:155]: Improve this message please.

File: src/main/java/org/folio/circulation/infrastructure/storage/sessions/PatronActionSessionRepository.java [hunk 3]
```diff
@@ -142,6 +146,7 @@ private JsonObject mapToJson(PatronSessionRecord patronSessionRecord) {
       .collect(Collectors.toCollection(HashSet::new));
 
     if (patronIds.isEmpty()) {
+      log.info("Patrons ids were not presented in expired sessions");
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/infrastructure/storage/sessions/PatronActionSessionRepository.java:149]:
```suggestion
      log.info("No patron IDs found. Doing nothing.");
```

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java [hunk 4]
```diff
@@ -43,12 +75,13 @@ public void expiredEndSessionAfterCheckOut() {
 
     expiredEndSessionClient.create(new EndSessionBuilder()
       .withPatronId(patronId)
-      .withActionType("Check-out"));
+      .withActionType(CHECK_OUT));
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/loans/EndExpiredPatronActionSessionTests.java:78]: Extract method?

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java [hunk 5]
```diff
@@ -207,7 +242,7 @@ public void patronsHaveSessionsAndAllShouldBeExpiredByTimeout() {
         .equals(PatronActionType.CHECK_IN.getRepresentation()))
       .map(session -> session.getString(PATRON_ID))
       .forEach(patronId -> expiredEndSessionClient.create(new EndSessionBuilder()
-        .withPatronId(patronId).withActionType("Check-in")));
+        .withPatronId(patronId).withActionType(CHECK_IN)));
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/loans/EndExpiredPatronActionSessionTests.java:245]: Add line break.

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java [hunk 6]
```diff
@@ -221,12 +256,13 @@ public void patronsHaveSessionsAndAllShouldBeExpiredByTimeout() {
         .equals(PatronActionType.CHECK_OUT.getRepresentation()))
       .map(session -> session.getString(PATRON_ID))
       .forEach(patronId -> expiredEndSessionClient.create(new EndSessionBuilder()
-        .withPatronId(patronId).withActionType("Check-out")));
+        .withPatronId(patronId).withActionType(CHECK_OUT)));
```
Comment (OleksandrVidinieiev) [inline src/test/java/api/loans/EndExpiredPatronActionSessionTests.java:259]: Add line break.

File: src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionService.java (hunk 2 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/domain/notice/session/PatronActionSessionService.java]: Improved

File: src/main/java/org/folio/circulation/infrastructure/storage/sessions/PatronActionSessionRepository.java (hunk 3 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/circulation/infrastructure/storage/sessions/PatronActionSessionRepository.java]: done

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java (hunk 4 above)
Comment (roman-barannyk) [reply src/test/java/api/loans/EndExpiredPatronActionSessionTests.java]: extracted

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java (hunk 5 above)
Comment (roman-barannyk) [reply src/test/java/api/loans/EndExpiredPatronActionSessionTests.java]: Extracted to separate method

File: src/test/java/api/loans/EndExpiredPatronActionSessionTests.java (hunk 6 above)
Comment (roman-barannyk) [reply src/test/java/api/loans/EndExpiredPatronActionSessionTests.java]: Extracted to separate method

**Comment by roman-barannyk** [comment]
I have created the new story that fully corresponds this issue. It's CIRC-628. I will close this PR and create the new one under the new ticket: https://github.com/folio-org/mod-circulation/pull/628

**Pull request closed without merging**